include/dt-bindings/deprecate.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2019 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_DT_BINDINGS_DEPRECATE_H_
+#define ZEPHYR_INCLUDE_DT_BINDINGS_DEPRECATE_H_
+
+#if defined(__GNUC__)
+#define __DT_DEPRECATED_MACRO _Pragma("GCC warning \"DTS Macro is deprecated\"")
+#endif
```

Comment:
Doesn't there have to be an `#else` branch that defines it to an empty token sequence so it doesn't break other toolchains?